### ISSUE TITLE
Backend fasta parser does not treat fasta headers with spaces correctly (according to spec)

### ISSUE THREAD

**Issue by corneliusroemer** [loculus-project/loculus#2923]
According to FASTA spec, the start of the header up to the first white space is the sequence id, everything after is description.

<details><summary>Current behavior and example</summary>

The backend currently seems to not follow the specs. Example, note sequence with header `custom5 1`, whose id should be parsed as `custom5` and be accepted.

`Body = {"type":"about:blank","title":"Unprocessable Entity","status":422,"detail":"Metadata file contains 1 submissionIds that are not present in the sequence file: custom5; Sequence file contains 1 submissionIds that are not present in the metadata file: custom5 1","instance":"/dummyOrganism/submit"}`

```
submissionId	date	region	country	division	host
custom4	2020-12-03	Europe	Switzerland	Zürich	Homo sapiens
custom0	2020-12-26	Europe	Switzerland	Bern	Homo sapiens
custom1	2020-12-15	Europe	Switzerland	Schaffhausen	Homo sapiens
custom2	2020-12-02	Europe	Switzerland	Bern	Homo sapiens
custom6	2020-12-16	Europe	Switzerland	Aargau
custom3		Europe	Switzerland	Schaffhausen	Homo sapiens
custom5	2020-12-23	Europe	Switzerland	Basel-Land	Homo sapiens
custom7	2XXXXX	Europe	Switzerland	Sankt Gallen	Homo sapiens
custom8	2020-12-16	Europe	Switzerland	Aargau	Homo sapiens
custom9	2020-12-01	Europe	Switzerland	Basel-Stadt	Homo sapiens
```

```fasta
>custom1
ACTG
>custom2
ACTG
>custom3
ACTG
>custom4
ACTG
ACTG
>custom7
ACT
ACT
>custom8
ACTG
>custom9
AC
>custom6
AC

>custom5 1
ACTG
>custom0
ACTG
```
</details>

[Slack discussion](https://loculus.slack.com/archives/C05F8GKR06R/p1739555769443819)

---

## Acceptance criteria

- When the backend parses FASTA, the sequence/submission ID should only go until the first whitespace. Everything after the first whitespace should be ignored.
- The behavior is documented (on the submit sequences page?).

**Comment by chaoran-chen** [comment]
What is the FASTA spec.? I found various definitions but couldn't identify a proper standard.

**Comment by corneliusroemer** [comment]
There's no single spec, but in many ways there's a rough consensus: https://www.ncbi.nlm.nih.gov/genbank/fastaformat/

Also see: https://www.biostars.org/p/11254/

**Comment by fhennig** [comment]
For documentation, I would like to link to a fasta file definition. Would you rather link to NCBI or the pacbiofileformats? @chaoran-chen @corneliusroemer

**Comment by theosanderson** [comment]
The pacbio one is laxer and therefore closer to what we're aiming at

**Comment by chaoran-chen** [comment]
If we just implement the ignoring of everything after the first whitespace (as @rneher suggested in [Slack](https://loculus.slack.com/archives/C05F8GKR06R/p1739613686200999?thread_ts=1739555769.443819&cid=C05F8GKR06R)), our implementation wouldn't be in exact accordance with either of the two definitions. The pacbio one seems closer but would be also more strict than ours.

**Comment by theosanderson** [comment]
👍  I wouldn't necessarily link to anything - but also not strongly against if it's helpful for the basics of how fastas work. Can also do https://en.wikipedia.org/wiki/FASTA_format

**Comment by chaoran-chen** [comment]
Agreed, Wikipedia as a general link about FASTA sounds good to me. And we should write in the documentation how exactly we are handling it.
